src/org/anjocaido/groupmanager/dataholder/WorldDataHolder.java

```diff
@@ -497,10 +492,8 @@ public boolean haveGroupsChanged() {
 	public void removeUsersChangedFlag() {
 
 		setUsersChanged(false);
-		synchronized (users) {
-			for (User u : getUsers().values()) {
-				u.flagAsSaved();
-			}

```

Comment:
I don't have an environment where I can do prolonged and extensive testing, so either this would need to be reverted or it would be published as a potentially highly unstable build